src/core/diags/factory_diags.cpp

```diff
@@ -557,6 +557,14 @@ otError Diags::ProcessCmd(uint8_t aArgsLength, char *aArgs[], char *aOutput, siz
 {
     otError error = OT_ERROR_NONE;
 
+#ifndef NDEBUG
```

Comment:
Looks good to me overall.
@bukepo , can you please explain the reason for `#ifndef NDEBUG`? Thanks.

Reply:
This code is just for testing or debug purpose, so guard this with `#ifndef NDEBUG` to remove it from release build.